### PR TITLE
Add unit tests for me.itzsomebody.radon.utils.ASMUtils

### PR DESCRIPTION
**Pull request by JohnLBergqvist**
I've analysed your codebase and noticed that `me.itzsomebody.radon.utils.ASMUtils` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

File: src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+import me.itzsomebody.radon.exceptions.RadonException;
+import me.itzsomebody.radon.utils.ASMUtils;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.objectweb.asm.tree.AbstractInsnNode;
+import org.objectweb.asm.tree.FrameNode;
+import org.objectweb.asm.tree.IntInsnNode;
```
Comment (petersear) [inline src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java]: Lines 11 and 12 could be swapped to maintain alphabetical order.

File: src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import org.objectweb.asm.tree.InsnNode;
+import org.objectweb.asm.tree.LabelNode;
+import org.objectweb.asm.tree.LdcInsnNode;
+import org.objectweb.asm.tree.LineNumberNode;
+import org.powermock.api.mockito.PowerMockito;
+
+public class ASMUtilsTest {
+
+  @Rule public final ExpectedException thrown = ExpectedException.none();
```
Comment (petersear) [inline src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java:20]: Add empty line after `@Rule`

**Review by freddytuxworth — APPROVED** [review]

File: src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  @Test
+  public void testIsReturn() {
+    Assert.assertFalse(ASMUtils.isReturn(0));
+    Assert.assertFalse(ASMUtils.isReturn(178));
+    Assert.assertTrue(ASMUtils.isReturn(176));
+  }
+
+  @Test
+  public void testIsInstruction() {
+    Assert.assertFalse(ASMUtils.isInstruction(PowerMockito.mock(FrameNode.class)));
```
Comment (petersear) [inline src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java]: These 4 lines are > 80 chars

File: src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    Assert.assertEquals(1L, ASMUtils.getLongFromInsn(new LdcInsnNode(1L)));
+
+    thrown.expect(RadonException.class);
+    ASMUtils.getLongFromInsn(new InsnNode(0));
+  }
+
+  @Test
+  public void testGetFloatFromInsn() throws Exception {
+    Assert.assertEquals(0.0f, ASMUtils.getFloatFromInsn(new InsnNode(11)), 0);
+    Assert.assertEquals(1.0f, ASMUtils.getFloatFromInsn(new LdcInsnNode(1.0f)), 0);
```
Comment (petersear) [inline src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java]: line > 80 chars

File: src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+    thrown.expect(RadonException.class);
+    ASMUtils.getFloatFromInsn(new InsnNode(0));
+  }
+
+  @Test
+  public void testGetDoubleFromInsn() throws Exception {
+    Assert.assertEquals(0.0, ASMUtils.getDoubleFromInsn(new InsnNode(14)), 0);
+    Assert.assertEquals(1.0, ASMUtils.getDoubleFromInsn(new LdcInsnNode(1.0)), 0);
```
Comment (petersear) [inline src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java]: line > 80 chars

**Review by petersear — CHANGES_REQUESTED** [review]

File: src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+    thrown.expect(RadonException.class);
+    ASMUtils.getLongFromInsn(new InsnNode(0));
+  }
+
+  @Test
+  public void testGetFloatFromInsn() throws Exception {
+    Assert.assertEquals(0.0f, ASMUtils.getFloatFromInsn(new InsnNode(11)), 0);
+    Assert.assertEquals(1.0f, ASMUtils.getFloatFromInsn(new LdcInsnNode(1.0f))
+            , 0);
```
Comment (petersear) [inline src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java]: I think the break should come after the comma...or the previous one...or both

File: src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+    thrown.expect(RadonException.class);
+    ASMUtils.getFloatFromInsn(new InsnNode(0));
+  }
+
+  @Test
+  public void testGetDoubleFromInsn() throws Exception {
+    Assert.assertEquals(0.0, ASMUtils.getDoubleFromInsn(new InsnNode(14)), 0);
+    Assert.assertEquals(1.0, ASMUtils.getDoubleFromInsn(new LdcInsnNode(1.0))
+            , 0);
```
Comment (petersear) [inline src/test/java/me/itzsomebody/radon/utils/ASMUtilsTest.java]: I think the break should come after the comma...or the previous one...or both

**Review by petersear — CHANGES_REQUESTED** [review]

**Review by petersear — APPROVED** [review]

**Pull request closed without merging**